test/helpers/kubectl.go

```diff
@@ -1726,6 +1730,30 @@ func (kub *Kubectl) CiliumPreFlightCheck() error {
 	return nil
 }
 
+// CiliumStatusPreFlightCheck validates the cilium status of all cilium pods
+// and returns an error if any of the pods report an unhealthy status
+func (kub *Kubectl) CiliumStatusPreFlightCheck() error {
+	ginkgoext.By("Checking that cilium status is healthy")
+
+	ciliumPods, err := kub.GetCiliumPods(KubeSystemNamespace)
+	if err != nil {
+		return err
+	}
+
+	for _, pod := range ciliumPods {
+		status := kub.CiliumExec(pod, "cilium status --all-health --all-nodes")
+		if !status.WasSuccessful() {
+			return fmt.Errorf("cilium-agent '%s': Cannot run cilium status: %s",
+				pod, status.OutputPrettyPrint())
+		}
+		if status.GetExitCode() != 0 {
```

Comment:
AFAIK the WasSuccessful is set if the status is 0, so these three lines does not make sense at all.